packages/core/src/amazonqTest/error.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { ToolkitError } from '../shared/errors'
+
+export const techinalErrorCustomerFacingMessage =
+    'I am experiencing technical difficulties at the moment. Please try again in a few minutes.'
+const defaultTestGenErrorMessage = 'Amazon Q encountered an error while generating tests. Try again later.'
+export class TestGenError extends ToolkitError {
+    constructor(
+        error: string,
+        code: string,
+        public statusCode: string,
+        public uiMessage: string
```

Comment:
I think we don't need uiMessage. The `ToolkitError.message` field is explicitly documented as "potentially shown to the user". You can store the lower-level message on `ToolkitError.details` https://github.com/aws/aws-toolkit-vscode/blob/e63079d3d3af18a3af27914e842d915f307b38b3/packages/core/src/shared/errors.ts#L161

This has far-reaching benefits. It allows all the existing architecture to treat these standard fields in a consistent way.

Reply:
Looks like details  has different type `readonly details?: Record<string, unknown>`

Reply:
Can you look at existing code and figure out a way to make it work with `details`